### PR TITLE
[12.0] account_mass_reconcile_transaction_ref: migration 

### PR DESCRIPTION
**Pull request by i-vyshnevska**
including commits from https://github.com/OCA/account-reconcile/pull/275

### CODE REVIEW

**Review by grindtildeath — APPROVED** [review]

**Review by jcoux — APPROVED** [review]
Technically seems good :+1:

File: account_mass_reconcile_transaction_ref/tests/test_account_reconcile_transaction_ref.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    def test_mass_reconcile_transaction_ref_vs_ref(self):
+        self.assertEqual(self.cust_invoice.state, 'open')
+        self.assertEqual(self.cust_invoice.transaction_id,
+                         'test_transaction_id')
+        bank_journal = self.env['account.journal'].search([
+            ('type', '=', 'sale')], limit=1)
+
+        # Create payment
+        payment = self.env['account.payment'].create({
```
Comment (Tonow-c2c) [inline account_mass_reconcile_transaction_ref/tests/test_account_reconcile_transaction_ref.py:51]: I think the payment creation can be factorized https://github.com/OCA/account-reconcile/pull/276/files#diff-2031f1d37eb509547481674f127dbd34R85

**Review by Tonow-c2c — APPROVED** [review]
LGTM

**Comment by grindtildeath** [comment]
I'm not sure if this will still be needed after #292

**Pull request merged**